getblocklab/block-lab
# Migrate `block-lab` namespace in `post_content`

#### Changes
* Adds PHP functions to migrate the entire site's `post_content` from the `block-lab/` block namespace to `genesis-custom-blocks/`
* There are more considerations after this PR, like a REST API or AJAX endpoint that the UI can use

#### Testing instructions
1. `wp db export # if you'd like, so you can restore to this backup of your DB `
2. `wp eval-file tests/bin/migrate-post-content.php`

<img width="546" alt="wp-cli-file" src="https://user-images.githubusercontent.com/4063887/84827506-2ff4c580-afea-11ea-9081-f282c205f8cf.png">

3. Expected: The WP-CLI command results in 'Success', like above
4. Go to the 'Revision' links shown above
5. Expected: The revision diff only involves `block-lab`. Nothing else should have changed:
<img width="1020" alt="exampled-diff-here" src="https://user-images.githubusercontent.com/4063887/84827765-8feb6c00-afea-11ea-893c-c7671c91d9f5.png">

6. Expected: If there's no `block-lab/` namespace in the content, the post should not be shown in the WP-CLI command output above, and there should of course be no revision.
7. It'd be great if you could go to 40-50 of those revision URLs to sanity-check them, assuming there are that many. It's tedious, but they're all just one revision ID higher than the next. 

8. To reverse this migration:
```sh
wp shell
( new Block_Lab\Blocks\Migration\Post_Content( 'genesis-custom-blocks', 'block-lab' ) )->migrate_all();
```
